### PR TITLE
[New Feature]add normalized_emb

### PR DESCRIPTION
**Pull request by jq**
# Description
add DynamicLayerNormalization since tf LayerNormalization doesn't support dynamic shape

Brief Description of the PR:

Fixes # (issue)

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Feature

# Checklist:

- [ ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/recommenders-addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running yapf
    - [ ] By running clang-format
- [ ] This PR addresses an already submitted issue for TensorFlow Recommenders-Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

File: demo/dynamic_embedding/movielens-1m-keras-with-horovod/movielens-1m-keras-with-horovod.py [hunk 1]
```diff
@@ -240,6 +241,100 @@ def get_kv_creator(mpi_size: int,
     return de.CuckooHashTableCreator(saver=saver)
 
 
+class DynamicLayerNormalization(LayerNormalization):
```
Comment (rhdong) [inline demo/dynamic_embedding/movielens-1m-keras-with-horovod/movielens-1m-keras-with-horovod.py:244]: Cool! It is really useful, and I recommend this layer could be a regular API, not only a demo.

File: tensorflow_recommenders_addons/dynamic_embedding/python/keras/layers/dynamic_layer_normalization.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import tensorflow as tf
+from tensorflow.keras.layers import LayerNormalization
+
+
+class DynamicLayerNormalization(LayerNormalization):
```
Comment (rhdong) [inline tensorflow_recommenders_addons/dynamic_embedding/python/keras/layers/dynamic_layer_normalization.py:5]: It is recommended to keep the name of `LayerNormalization` and just use a different prefix sort of `de.keras.layers.LayerNormalization`. Accordingly, the import statement needs to be modified to `from tensorflow.keras.layers import TFLayerNormalization`

File: tensorflow_recommenders_addons/dynamic_embedding/python/keras/layers/dynamic_layer_normalization_test.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import numpy as np
+
+import tensorflow as tf
+
+from tensorflow_recommenders_addons.dynamic_embedding.python.keras.layers.dynamic_layer_normalization import \
+    DynamicLayerNormalization
+
+
+class DynamicLayerNormalizationTest(tf.test.TestCase):
```
Comment (rhdong) [inline tensorflow_recommenders_addons/dynamic_embedding/python/keras/layers/dynamic_layer_normalization_test.py:9]: We'd better to have one case testing the scenario in which the new layer works with tfra API like the demo shows.

File: tensorflow_recommenders_addons/dynamic_embedding/python/keras/layers/dynamic_layer_normalization.py (hunk 2 above)
Comment (jq) [reply tensorflow_recommenders_addons/dynamic_embedding/python/keras/layers/dynamic_layer_normalization.py:5]: you mean from tensorflow.keras.layers import LayerNormalization as TFLayerNormalization ?

Comment (rhdong) [reply tensorflow_recommenders_addons/dynamic_embedding/python/keras/layers/dynamic_layer_normalization.py:5]: Correct! And using `de.keras.layers.LayerNormalization`

Comment (jq) [reply tensorflow_recommenders_addons/dynamic_embedding/python/keras/layers/dynamic_layer_normalization.py:5]: how about de.LayerNormalization?

**Review by rhdong — APPROVED** [review]
LGTM
